### PR TITLE
Guide: intro edits

### PR DESCRIPTION
**Pull request by mrhota**
This is mostly organizational and simplifying. However, I also removed prose which I took to be too basic given that we expect this to be read by programmers. Comments welcome.

cc @steveklabnik

### CODE REVIEW

File: src/doc/guide.md [hunk 1]
```diff
@@ -169,17 +160,17 @@ fn main() {
...
 
-These lines define a **function** in Rust. The `main` function is special:
-it's the beginning of every Rust program. The first line says "I'm declaring a
+These lines define a **function** in Rust. The `main` function is special: it's
+the beginning of every Rust program. The first line says "I'm declaring a
 function named `main`, which takes no arguments and returns nothing." If there
-were arguments, they would go inside the parentheses (`(` and `)`), and because
-we aren't returning anything from this function, we've dropped that notation
-entirely.  We'll get to it later.
+were arguments, they would go inside the parentheses, and because we aren't
```
Comment (mrhota) [inline src/doc/guide.md]: I'm happy to put them back in. But there's no ambiguity about what parentheses are. The punctuation there was distracting and kind of pointless.

File: src/doc/guide.md [hunk 2]
```diff
@@ -189,42 +180,42 @@ Next up is this line:
...
+those. Don't be scared of using macros. We'll get to the details eventually, but
+you'll just have to trust us for now.
 
 Next, `"Hello, world!"` is a **string**. Strings are a surprisingly complicated
 topic in a systems programming language, and this is a **statically allocated**
 string. We will talk more about different kinds of allocation later. We pass
-this string as an argument to `println!`, which prints the string to the
-screen. Easy enough!
+this string as an argument to `println!`, which prints the string and a linefeed
+character to the screen. Easy enough!
```
Comment (mrhota) [inline src/doc/guide.md]: "prints the string and a newline to the screen."

Newline is more correct and platform independent.

Comment (steveklabnik) [inline src/doc/guide.md]: Newline is almost never correct, it's either `\r\n` or `\n\r` depending on platform.

File: src/doc/guide.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/guide.md]: It's more consistent with the rest of the docs, which do mention words for symbols upon seeing them the first time.

File: src/doc/guide.md (hunk 2 above)
Comment (mrhota) [inline src/doc/guide.md]: I don't mean to argue, but a few sources from the wikipedia page below refer to the platform-specific character sequence used to a new line, not \r or \n specifically.

See also: https://en.wikipedia.org/wiki/Newline
"... a special character or sequence of characters signifying the end of a line of text. ... a newline in character encoding can be defined as LF and CR combined into one (LF+CR, LFCR, CR+LF, CRLF)."

Comment (steveklabnik) [inline src/doc/guide.md]: You know what, this is just a total brain fart on my part. I'm not sure why I went "CR + LF" and then suddenly went "newline." My bad. :(

**Comment by frewsxcv** [comment]
Needs another rebase

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to reopen with a rebase!

**Pull request closed without merging**